runtime/vam/expr/queryexpr.go

```diff
@@ -0,0 +1,94 @@
+package expr
+
+import (
+	"github.com/brimdata/super"
+	"github.com/brimdata/super/runtime"
+	"github.com/brimdata/super/vector"
+	"github.com/brimdata/super/vector/bitvec"
+	"github.com/brimdata/super/zcode"
+)
+
+type QueryExpr struct {
+	rctx   *runtime.Context
+	puller vector.Puller
+	cached vector.Any
+}
+
+func NewQueryExpr(rctx *runtime.Context, puller vector.Puller) *QueryExpr {
+	return &QueryExpr{rctx: rctx, puller: puller}
+}
+
+func (q *QueryExpr) Eval(this vector.Any) vector.Any {
+	if q.cached == nil {
+		q.cached = q.exec(this)
+	}
+	switch vec := q.cached.(type) {
+	case *vector.Const:
+		return vector.NewConst(vec.Value(), this.Len(), bitvec.Zero)
+	default:
+		if this.Len() > 1 {
+			// This is an array so just create a view that repeats this.Len().
+			return vector.Pick(vec, make([]uint32, this.Len()))
+		}
+		return vec
+	}
+
+}
+
+func (q *QueryExpr) exec(this vector.Any) vector.Any {
```

Comment:
Nit: We don't need really need `this` here.

```suggestion
func (q *QueryExpr) exec(length uint32) vector.Any {
```